config/initializers/activity_csv_columns.rb

```diff
@@ -2,3 +2,18 @@
   File.read("#{Rails.root}/config/activity_csv_columns.yml"),
   symbolize_names: true
 )[:columns].freeze
+
+level_b_non_oda_attributes = []
+level_b_oda_attributes = []
+level_c_d_non_oda_attributes = []
+level_c_d_oda_attributes = []
+
+ACTIVITY_CSV_COLUMNS.each do |attribute, column|
+  level_b_non_oda_attributes << attribute if column.dig(:inclusion, :level_b, :ispf_non_oda)
+  level_b_oda_attributes << attribute if column.dig(:inclusion, :level_b, :ispf_oda)
+  level_c_d_non_oda_attributes << attribute if column.dig(:inclusion, :level_c_d, :ispf_non_oda)
+  level_c_d_oda_attributes << attribute if column.dig(:inclusion, :level_c_d, :ispf_oda)
+end
+
+INVALID_LEVEL_B_ISPF_NON_ODA_ATTRIBUTES = level_b_oda_attributes - level_b_non_oda_attributes
```

Comment:
Shall we `.freeze` these to be extra safe?

Reply:
Sure!

Reply:
Does this approach more generally seem sensible to you? If so, I'll squash the commits

Reply:
I think so - I sort of wish there was a way that didn't involve looping all over and subtracting (like if we could just do something like `columns.where(level_b: true, ispf_oda: true, ispf_non_oda: false)`, but I can't think of a neat way of doing that with the way our hash is structured right now. Let's go with this for now.

Reply:
Yeah, `filtered_csv_column_headings` is essentially that kind of interface but for headers, not attributes. I suppose we could make an almost identical `filtered_csv_column_attributes` (or something more accurate) but the only use case right now is working out the difference